azkaban-common/src/main/java/azkaban/imagemgmt/daos/RampRuleDaoImpl.java

```diff
@@ -93,9 +96,50 @@ public int addRampRule(final ImageRampRule rampRule) {
     }
   }
 
+  /**
+   * update column owners from table ramp_rules along with modification metadata.
+   *
+   * @param newOwners
+   * @param ruleName
+   * @param modifiedBy
+   * @return int - id of the DB entry
+   * @throws ImageMgmtDaoException
+   */
+  @Override
+  public int updateOwnerships(final String newOwners, final String ruleName, final String modifiedBy) {
+    try {
+      // validation should already done
+      return this.databaseOperator.update(UPDATE_RULE_OWNERSHIP,
+          newOwners, modifiedBy, Timestamp.valueOf(LocalDateTime.now()), ruleName);
+    } catch (SQLException e) {
+      LOG.error("Error in updating ownership " + newOwners);
+      throw new ImageMgmtDaoException(ErrorCode.INTERNAL_SERVER_ERROR,
+          "Error in updating ownership with " + e.getMessage());
+    }
+  }
+
+  /**
+   * Query table ramp_rules to get owners of Ramp rule.
+   * Serves as the source of truth for rule ownership management.
+   *
+   * @param ruleName - ruleName in {@see ImageRampRule}
+   * @return owners of the ramp rule.
+   */
   @Override
-  public String getOwners(final String ruleName) {
-    return null;
+  public Set<String> getOwners(final String ruleName) {
```

Comment:
the above method may not need test, but this method worth add some simple tests

Reply:
I changed the service implementation code to cover updateOwners which would be using this Dao.getOwner.
Unless we setup test DB in for integration test otherwise it's useless only mock the behavior in unit test.